### ISSUE TITLE
SC2034 False positive on valid bash assignment of variable without ((..)).

### ISSUE THREAD

**Issue by TheCayman** [koalaman/shellcheck#1723]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2034
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

#!/bin/bash
str1="a piece of text"
str2=str1
echo "${str2}"

#### Here's what shellcheck currently says:
```
Line 2	SC2034: str1 appears unused. Verify use (or export if used externally).
```
#### Here's what I wanted or expected to see:
a piece of text

This is perfectly valid bash and does produce the expected output. 
Shellcheck will not print an error message if I either replace the line str2=str1 with:

((str2=str1))
or
str2=$str1

This looks to be a possible duplicate of #1705

**Comment by gabrielelana** [comment]
When executing your script the output is `str1`, so seems legit to tell you that the variable `str1` is not used or am I missing something?

**Comment by TheCayman** [comment]
I oversimplified the code before creating this Issue. I am terribly sorry about that.

Both variables actually need to be declared as integers (as opposed to strings) for this problem to surface. I have changed the code to properly fail and show the problem.

**Comment by matthewpersico** [comment]
Suggest title change:

SC2034 False positive on valid bash assignment of integer variable without ((..)) or $
